### ISSUE TITLE
[Feature Request] Add Buku Support

### ISSUE THREAD

**Issue by gregf** [dhruvmanila/telescope-bookmarks.nvim#7]
[Buku](https://github.com/jarun/buku) is a command line bookmark manager that also uses sqlite. It would be great to have it working as well.

**Comment by dhruvmanila** [comment]
Can you try out the [`feat/buku`](https://github.com/dhruvmanila/telescope-bookmarks.nvim/tree/feat/buku) branch?

**Comment by dhruvmanila** [comment]
Can you try out the `feat/buku` branch?

**Comment by gregf** [comment]
Wow, thanks for doing this so quickly. It brings up my bookmarks currently, but I can't get it to open anything in Firefox. I set `      url_open_command = 'xdg-open'`. Which works on the command line if I pass a URL to `xdg-open`. Within the plugin though, nothing appears to be happening when I press enter on a bookmark.

**Comment by dhruvmanila** [comment]
Hmm, that's weird as it's working fine on my Mac. Can you try it with any browser which contains some bookmarks? Also, can you try it with a minimal config using the template below?
1. Open Neovim with `nvim -n -u path/to/minimal.lua`
2. Wait for everything to be installed. It will print `Ready!` once its done
3. Use `:Telescope bookmarks`

```lua
-- Load the 'runtime/' files
vim.cmd [[set runtimepath=$VIMRUNTIME]]

-- Originally, `packpath` contains a lot of path to search into which also
-- includes the `~/.config/nvim` directory. Now, if we open Neovim, the files
-- in the `plugin/`, `ftplugin/`, etc. directories will be loaded automatically.
--
-- We will set the value of `packpath` to contain only our testing directory to
-- avoid loading files from our config directory.
--
--     $ nvim -nu minimal.lua
vim.cmd [[set packpath=/tmp/nvim/site]]

local package_root = "/tmp/nvim/site/pack"
local packer_install_path = package_root .. "/packer/start/packer.nvim"

local function load_plugins()
  require("packer").startup {
    {
      "wbthomason/packer.nvim",
      -- Add plugins to test...
      "nvim-telescope/telescope.nvim",
      "nvim-lua/plenary.nvim",
      "tami5/sqlite.lua",
      { "dhruvmanila/telescope-bookmarks.nvim", branch = "feat/buku" },
    },
    config = {
      package_root = package_root,
      compile_path = packer_install_path .. "/plugin/packer_compiled.lua",
      display = { non_interactive = true },
    },
  }
end

_G.load_config = function()
  -- Add the necessary `init.lua` settings which could include the setup
  -- functions for the plugins...
  require("telescope").setup {
    extensions = {
      bookmarks = {
        selected_browser = "buku",
      },
    },
  }

  require("telescope").load_extension "bookmarks"
end

if vim.fn.isdirectory(packer_install_path) == 0 then
  print "Installing plugins and dependencies..."
  vim.fn.system {
    "git",
    "clone",
    "--depth=1",
    "https://github.com/wbthomason/packer.nvim",
    packer_install_path,
  }
end

load_plugins()
require("packer").sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]]
```

**Comment by dhruvmanila** [comment]
Also, once you open the picker, can you see the bookmark name and its URL?

**Comment by gregf** [comment]
Ok I made your minimal.lua loaded it up like you said above. Once it was all loaded I ran `:Telescope bookmarks` and got the following error, which I did not get in my config.

```
E5108: Error executing lua ...e-bookmarks.nvim/lua/telescope/_extensions/bookmarks.lua:84: attempt to concatenate field 'name' (a nil value)
stack traceback:
        ...e-bookmarks.nvim/lua/telescope/_extensions/bookmarks.lua:84: in function 'entry_maker'
        ...scope.nvim/lua/telescope/finders/async_static_finder.lua:17: in function 'new_table'
        ...e-bookmarks.nvim/lua/telescope/_extensions/bookmarks.lua:77: in function <...e-bookmarks.nvim/lua/telescope/_extensions/bookmarks.lua:41>
        ...ck/packer/start/telescope.nvim/lua/telescope/command.lua:184: in function 'run_command'
        ...ck/packer/start/telescope.nvim/lua/telescope/command.lua:241: in function 'load_command'
        [string ":lua"]:1: in main chunk
```

In my config I do see the url and the name btw.

**Comment by dhruvmanila** [comment]
Ok, I think I get what the problem is: Can you show me the output of `buku --print 1-5 --json`?

**Comment by dhruvmanila** [comment]
Hey, can you try out the branch again, I've force pushed it. Also, please provide me with the version of `buku`.

If it's still not working, then can you give me the schema of the database?
1. Locate the database file, it should most likely be `~/.local/share/buku/bookmarks.db`
2. Open it with `sqlite3 bookmarks.db` (you might have to install the command)
3. Give me the output of `.schema` command

Also, if possible, provide me the output for the following query:
```sql
SELECT * FROM bookmarks LIMIT 10;
```

**Comment by gregf** [comment]
@dhruvmanila Hey, thank you so much. Working perfectly now. I can still provide a schema, if you would find that useful, let me know. Do you have a donate button I can find? You rock.

Actually let's save a step, here's the schema just in case.

```
CREATE TABLE bookmarks (id integer PRIMARY KEY, URL text NOT NULL UNIQUE, metadata text default '', tags text default ',', desc text default '', flags integer default 0);
```

I'll let you close the issue when you feel fit. Thank you again.

**Comment by dhruvmanila** [comment]
> Hey, thank you so much. Working perfectly now.

Awesome!

> I can still provide a schema, if you would find that useful, let me know.

It's not needed anymore.

> Do you have a donate button I can find? You rock.

Thanks, but I haven't set that up yet.

I'll merge the branch and close this.

**Comment by dhruvmanila** [comment]
Closed by 06a6865105c39d6305c27aa96c00d4629d666aaa
